engine/src/main/java/org/terasology/entitySystem/event/internal/EventSystemImpl.java

```diff
@@ -352,11 +350,9 @@ public void send(EntityRef entity, Event event, Component component) {
             if (handlers != null) {
                 List<EventHandlerInfo> eventHandlers = Lists.newArrayList(handlers.get(component.getClass()));
                 Collections.sort(eventHandlers, priorityComparator);
-                for (EventHandlerInfo eventHandler : eventHandlers) {
```

Comment:
This method is called very often. I would leave its code as-is, because I think that the simple iteration is more performant than the conversion and lambda-based filtering.